crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1890,35 +1899,45 @@ impl<'src> Parser<'src> {
 
     /// Parses a `yield from` expression.
     ///
+    /// This method should not be used directly. Use [`Parser::parse_yield_expression`]
+    /// even when parsing a `yield from` expression.
+    ///
     /// See: <https://docs.python.org/3/reference/expressions.html#yield-expressions>
     fn parse_yield_from_expression(&mut self, start: TextSize) -> Expr {
-        // The grammar rule to use here is `expression` but we're using `expressions`
-        // for better error recovery.
-        let parsed_expr = self.parse_expression_list();
-
-        match &parsed_expr.expr {
-            Expr::Starred(ast::ExprStarred { value, .. }) => {
-                self.add_error(
-                    ParseErrorType::OtherError(
-                        "starred expression is not allowed in a `yield from` statement".to_string(),
-                    ),
-                    value.as_ref(),
-                );
+        // Grammar:
+        //     'yield' 'from' expression
+        //
+        // Here, a tuple expression isn't allowed without the parentheses. But, we
+        // allow it here to report better error message.
+        //
+        // Now, this also solves another problem. Take the following example:
+        //
+        // ```python
+        // yield from x, y
+        // ```
+        //
+        // If we didn't use the `parse_expression_list` method here, the parser
+        // would have stopped at the comma. Then, the outer expression would
+        // have been a tuple expression with two elements: `yield from x` and `y`.
+        let expr = self.parse_expression_list().expr;
+
+        match &expr {
+            Expr::Starred(_) => {
+                self.add_error(ParseErrorType::StarredExpressionUsage, &expr);
             }
             Expr::Tuple(tuple) if !tuple.parenthesized => {
                 self.add_error(
                     ParseErrorType::OtherError(
-                        "unparenthesized tuple is not allowed in a `yield from` statement"
-                            .to_string(),
+                        "unparenthesized tuple cannot be used here".to_string(),
```

Comment:
It might be valuable to provide context such as "in a `yield from` statement" in this message. Valuable in the sense that the parser provides an explanation that it's not allowed specifically in _yield from_ statement. But, I think this might be the job of a future linter diagnostic group for syntax errors.